src/passes/OptimizeInstructions.cpp

```diff
@@ -294,6 +294,23 @@ struct OptimizeInstructions
           return sub;
         }
       }
+      {
+        // eqz((signed)x % C_pot)   =>   eqz(x & (C_pot - 1))
+        Const* c;
+        Unary* eqz;
+        Binary* inner;
+        if (matches(
+              curr,
+              unary(&eqz,
+                    Abstract::EqZ,
+                    binary(&inner, Abstract::RemS, any(), constant(&c))))) {
+          if (IsPowerOf2((uint64_t)c->value.getInteger())) {
+            inner->op = Abstract::getBinary(c->type, Abstract::And);
+            c->value = c->value.sub(Literal::makeFromInt32(1, c->type));
+            return eqz;
```

Comment:
This can just be `return curr;` and the `eqz` variable can be removed. Also you can use `ival` instead of `constant` and rename `inner` to `bin` to make the pattern more compact.